webpages/settings/components/modal.js

```diff
@@ -0,0 +1,35 @@
+export default async function ({ template }) {
+  const Modal = Vue.extend({
+    props: ["isOpen", "title"],
+    template,
+    data() {
+      return {
+        canCloseOutside: false,
+      };
+    },
+    watch: {
+      isOpen(value) {
+        if (value) {
+          setTimeout(() => {
+            this.canCloseOutside = true;
+          }, 100);
+        } else {
+          this.canCloseOutside = false;
+        }
+      },
+    },
+    methods: {
+      msg(...params) {
+        return this.$root.msg(...params);
+      },
+    },
+    events: {
+      clickOutside(e) {
+        if (this.isOpen && this.canCloseOutside && e.isTrusted) {
```

Comment:
Do we need isTrusted check?

Reply:
This is probably code from stackoverflow?

Reply:
It was added in #1136 to ignore the click event from the hidden file input.

Reply:
ah, makes sense